receiver/statsdreceiver/internal/transport/udp_server.go

```diff
@@ -31,60 +27,14 @@ func NewUDPServer(transport Transport, address string) (Server, error) {
 	}
 
 	return &udpServer{
-		packetConn: conn,
-		transport:  transport,
+		packetServer: packetServer{
+			packetConn: conn,
+			transport:  transport,
+		},
 	}, nil
 }
 
-// ListenAndServe starts the server ready to receive metrics.
-func (u *udpServer) ListenAndServe(
```

Comment:
This refactoring makes sense but maybe would be better if it's done in a standalone/different PR instead of mixing it with this addition: https://github.com/open-telemetry/opentelemetry-collector/blob/main/CONTRIBUTING.md#refactoring-work

Reply:
Hmm I see. I would argue that this refactoring work is necessary for this functionality. Otherwise, we'd be unnecessarily duplicating code for the new functionality.

If you feel strongly about this, I can make the change.

Reply:
Yeap, I see. I'm fine keeping it as is.